infrastructure/ssz/src/test/java/tech/pegasys/teku/infrastructure/ssz/collections/SszBitlistTest.java

```diff
@@ -394,4 +402,61 @@ void testBitEmptyListSsz(final SszBitlist bitlist) {
     SszBitlist emptyList1 = bitlist.getSchema().sszDeserialize(Bytes.of(1));
     assertThat(emptyList1).isEmpty();
   }
+
+  @ParameterizedTest
+  @MethodSource("fromBytesTestCases")
+  void testFromBytes(final SszBitlist bitlist, final Bytes serialized) {
+    SszBitlist deserialized = bitlist.getSchema().fromBytes(serialized);
+
+    assertThat(deserialized).isEqualTo(bitlist);
+    assertThat(deserialized.size()).isEqualTo(bitlist.size());
+    assertThatIntCollection(deserialized.getAllSetBits()).isEqualTo(bitlist.getAllSetBits());
+    assertThat(deserialized.hashTreeRoot()).isEqualTo(bitlist.hashTreeRoot());
+  }
+
+  @ParameterizedTest
+  @MethodSource("fromHexStringTestCases")
+  void testFromHexString(final SszBitlist bitlist, final String hexString) {
+    SszBitlist deserialized = bitlist.getSchema().fromHexString(hexString);
+
+    assertThat(deserialized).isEqualTo(bitlist);
+    assertThat(deserialized.size()).isEqualTo(bitlist.size());
+    assertThatIntCollection(deserialized.getAllSetBits()).isEqualTo(bitlist.getAllSetBits());
+    assertThat(deserialized.hashTreeRoot()).isEqualTo(bitlist.hashTreeRoot());
+  }
+
+  @Test
+  public void testFromHexStringEdgeCases() {
```

Comment:
I think this would be even better if we break this general edge cases into individual tests?

Reply:
good point - updated!